pytest_html/plugin.py

```diff
@@ -112,8 +113,10 @@ def __init__(self, outcome, report, contained_html, logfile):
             self.contained_html = contained_html
             self.logfile = logfile
 
+            extra_index = 1
             for extra in getattr(report, 'extra', []):
-                self.append_extra_html(extra)
+                self.append_extra_html(extra, extra_index, test_index)
```

Comment:
I understand why you went with the number of tests with a specific outcome, but I think using `report.rerun` would be a better solution. Your solution will mean a different hash will be generated if the tests run in a different order, or if a test has a different outcome. By using the run number for the individual test we'll be able to rewrite files in the assets directory on subsequent runs, causing less junk to build up. It also means we can avoid passing the additional integer for every `TestResult` object, as we already have the `report` object.